crates/uv/src/commands/workspace/dir.rs

```diff
@@ -0,0 +1,47 @@
+use std::fmt::Write;
+use std::path::Path;
+
+use anyhow::{Result, bail};
+
+use uv_cli::WorkspaceDirArgs;
+use uv_fs::PortablePathBuf;
+use uv_preview::{Preview, PreviewFeatures};
+use uv_warnings::warn_user;
+use uv_workspace::{DiscoveryOptions, Workspace, WorkspaceCache};
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Print the path to the workspace dir
+pub(crate) async fn dir(
+    args: &WorkspaceDirArgs,
+    project_dir: &Path,
+    preview: Preview,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_enabled(PreviewFeatures::WORKSPACE_METADATA) {
```

Comment:
I think this should be a separate preview flag so we can stabilize things separately